src/pages/new-chat/new-chat.ts

```diff
@@ -9,22 +12,30 @@ import { User } from 'api/models/whatsapp-models';
   templateUrl: 'new-chat.html'
 })
 export class NewChatComponent implements OnInit {
-  users;
+  searchPattern = '';
   senderId: string;
+  users: Observable<User[]>;
 
   constructor(
-    public navCtrl: NavController,
-    public viewCtrl: ViewController,
-    public alertCtrl: AlertController
+    private navCtrl: NavController,
+    private viewCtrl: ViewController,
+    private alertCtrl: AlertController
   ) {
     this.senderId = Meteor.userId();
   }
 
   ngOnInit() {
-    MeteorObservable.subscribe('users').subscribe(() => {
-      MeteorObservable.autorun().subscribe(() => {
-        this.users = this.findUsers().zone();
-      });
+    this.findContacts().then((contacts) => {
```

Comment:
What about a case of rejection from the native `Contacts` Promise? this might happen if there are no permissions for contacts...